### ISSUE TITLE
Consider specifying AXTitleUIElement as exposed in more cases

### ISSUE THREAD

**Issue by twilco** [w3c/html-aam#611]
[The core-aam spec lists AXTitleUIElement as exposed for `aria-labelledby`](https://w3c.github.io/aria/core-aam/#ariaLabelledBy) when there is one pointed-to element (not exposed when there are multiple pointed-to elements).

However, there are additional cases where WebKit exposes `AXTitleUIElement` that are not in the spec:
* `<label>` relationships (explicit with `for` and implicit via ancestry): https://w3c.github.io/aria/html-aam/#el-label
* `<fieldset>` + `<legend>`: https://w3c.github.io/aria/html-aam/#el-fieldset
* For cells within a layout table (vs. an "accessibility table", as [determined by heuristics](https://github.com/WebKit/WebKit/blob/038b254e879cfdd51c5f7443fb4dde51a48081cc/Source/WebCore/accessibility/AccessibilityNodeObject.cpp#L2247)), the `<th>` in column zero of the associated row is exposed as the `AXTitleUIElement`. This one is far less clear-cut as something that should be specified.

There's also `<caption>` + `<table>` (https://w3c.github.io/aria/html-aam/#el-caption), where both WebKit does not expose `AXTitleUIElement`, and the spec doesn't say it should (but probably ought to). There may have been a copy-paste error in this one, as there is an `AX` row, but it references `ATK_RELATION_LABEL_FOR`.

Example markup for the table cell case in which WebKit exposes `AXTitleUIElement`:

```html
<table role="presentation">
  <tr><th id="h1">Header one</th><td>Exposes #h1</td></tr>
  <tr><th id="h2">Header two</th><td>Exposes #h2</td></tr>
</table>
```
